src/libraries/System.ObjectModel/tests/TrimmingTests/TypeDescriptionProviderAttributeCtorTest.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.ComponentModel;
+using System.Xml.Linq;
+
+class Program
+{
+    /// <summary>
+    /// Tests that the public parameterless constructors of types passed to the
+    /// constructors of System.ComponentModel.TypeDescriptionProviderAttribute
+    /// property are preserved when needed in a trimmed application.
+    /// </summary>
+    static int Main(string[] args)
+    {
+        // Test that public parameterless ctor of the Type argument here is preserved.
+        Type providerType = typeof(MyTypeDescriptionProvider);
+        TypeDescriptionProviderAttribute attr = new TypeDescriptionProviderAttribute(providerType);
+        object obj = Activator.CreateInstance(providerType);
```

Comment:
Calling `Activator.CreateInstance(providerType)` is going to preserve the constructor by itself.

Instead, you need to say something like:

```
                Type providerType = Type.GetType(attr.TypeName);
                if (providerType != null && typeof(TypeDescriptionProvider).IsAssignableFrom(providerType))
                {
                    TypeDescriptionProvider prov = (TypeDescriptionProvider)Activator.CreateInstance(providerType);
```

like the product code does: 

[https://github.com/dotnet/runtime/blob/bd6cbe3642f51d70839912a6a666e5de747ad581/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs#L332-L339](https://github.com/dotnet/runtime/blob/bd6cbe3642f51d70839912a6a666e5de747ad581/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs#L332-L339)